pkg/snapshot/snapshotter/snapshotter.go

```diff
@@ -241,9 +243,23 @@ func (ssr *Snapshotter) takeFullSnapshot() (*snapstore.Snapshot, error) {
 				Message: fmt.Sprintf("failed to create etcd snapshot: %v", err),
 			}
 		}
+
+		// compressionSuffix is useful in backward compatibility(restoring from non-compressed snapshots).
+		// it is also helpful in inferring which compression Policy to be used to decompress the snapshot.
+		compressionSuffix := compressor.GetCompressionSuffix(ssr.compressionConfig.Enabled, ssr.compressionConfig.CompressionPolicy)
 		ssr.logger.Infof("Successfully opened snapshot reader on etcd")
-		s := snapstore.NewSnapshot(snapstore.SnapshotKindFull, 0, lastRevision)
+		s := snapstore.NewSnapshot(snapstore.SnapshotKindFull, 0, lastRevision, compressionSuffix)
+
 		startTime := time.Now()
+
+		// if compression is enabled
+		//    then compress the snapshot.
+		if ssr.compressionConfig.Enabled {
+			ssr.logger.Info("start the Compression of full snapshot")
+			rc = compressor.CompressSnapshot(rc, ssr.compressionConfig.CompressionPolicy)
```

Comment:
If `CompressSnapshot` returns error (if you add the default case) then it will have to be handled here.